src/cli.ts

```diff
@@ -0,0 +1,378 @@
+  if (!options.endCommit) {
+    throw new Error("Missing required argument: --end (-e)");
+  }
+  if (options.files.length === 0) {
+    throw new Error(
+      "Missing required argument: --files (-f) or --file-list (-F)",
+    );
+  }
+}
+
+/**
+ * Print help message.
+ */
+export function printHelp(): void {
+  console.log(`
+pickdiff - Generate diffs between Git commits for specific files
+
+Usage:
+  pickdiff [options]
+
+Required Options:
+  -s, --start <commit>     Start commit (base commit)
+  -e, --end <commit>       End commit (target commit)
+  -f, --files <files>      Comma-separated list of files to diff
+  -F, --file-list <path>   Path to file containing list of files (one per line)
+
+Optional:
+  -r, --repo <path>        Repository path (default: current directory)
+  -c, --context <lines>    Number of context lines (default: 3)
+  -o, --output <format>    Output format: 'unified' or 'markdown' (default: unified)
+  -h, --help               Show this help message
+  -v, --version            Show version
+
+Examples:
+  pickdiff -s HEAD~5 -e HEAD -f src/index.ts,src/utils.ts
+  pickdiff --start abc123 --end def456 --file-list files.txt --output markdown
+  pickdiff -r /path/to/repo -s main -e feature-branch -f README.md
+`);
+}
+
+/**
+ * Print version information.
+ */
+export function printVersion(): void {
+  // Read version from package.json
+  try {
+    const packageJsonPath = path.join(__dirname, "..", "package.json");
+    const packageJson = JSON.parse(fs.readFileSync(packageJsonPath, "utf-8"));
+    console.log(`pickdiff v${packageJson.version}`);
+  } catch {
+    console.log("pickdiff (version unknown)");
+  }
+}
+
+/**
+ * Format diff output as unified diff.
+ * @param diffs The diff result containing file diffs
+ * @returns Formatted unified diff string
+ */
+export function formatUnifiedDiff(diffs: DiffResult): string {
+  const lines: string[] = [];
+
+  for (const file of Object.keys(diffs.diffs)) {
+    const diffLines = diffs.diffs[file];
+
+    // Add file header
+    lines.push(`--- a/${file}`);
+    lines.push(`+++ b/${file}`);
+
+    // Check for no changes marker
+    if (diffLines.length === 1 && diffLines[0].content === "NO_CHANGES") {
+      lines.push("(no changes)");
+      lines.push("");
+      continue;
+    }
+
+    // Output diff lines
+    for (const diffLine of diffLines) {
+      lines.push(diffLine.content);
+    }
+    lines.push("");
+  }
+
+  // Add excluded files warning if any
+  if (diffs.excludedFiles.length > 0) {
+    lines.push("");
+    lines.push("# Excluded files (not found in end commit):");
+    for (const file of diffs.excludedFiles) {
+      lines.push(`#   ${file}`);
+    }
+  }
+
+  return lines.join("\n");
+}
+
+/**
+ * Format diff output as markdown.
+ * @param diffs The diff result containing file diffs
+ * @param options CLI options for metadata
+ * @returns Formatted markdown string
+ */
+export function formatMarkdown(diffs: DiffResult, options: CliOptions): string {
```

Comment:
Does the functionality in `formatMarkdown()` overlap with `markdown.ts`? Can we reuse some of that code?